test/stdlib/Ractor_test.rb

```diff
@@ -0,0 +1,173 @@
+require_relative "test_helper"
+
+if RUBY_VERSION < '3'
+  warn 'Ractor is not available on Ruby 2🐫 Skip the test'
+  return
+end
+
+class RactorSingletonTest < Test::Unit::TestCase
+  include TypeAssertions
+
+  testing "singleton(::Ractor)"
+
+  def test_count
+    assert_send_type "() -> Integer",
+                     Ractor, :count
+  end
+
+  def test_current
+    assert_send_type "() -> Ractor",
+                     Ractor, :current
+  end
+
+  def test_main
+    assert_send_type "() -> Ractor",
+                     Ractor, :main
+  end
+
+  def test_make_shareable
+    assert_send_type "(String) -> String",
+                     Ractor, :make_shareable, 'foo'
+    assert_send_type "(String, copy: true) -> String",
+                     Ractor, :make_shareable, 'foo', copy: true
+    assert_send_type "(String, copy: false) -> String",
+                     Ractor, :make_shareable, 'foo', copy: false
+    assert_send_type "(String, copy: nil) -> String",
+                     Ractor, :make_shareable, 'foo', copy: nil
+  end
+
+  def test_new
+    # TODO: it raises an error because the proc is not isolated
```

Comment:
I guess we need to make `assert_send_type` Ractor safe to avoid this error, and I guess it is not easy. So I skipped the test case.